src/baselaplace.jl

```diff
@@ -35,6 +35,9 @@ function get_params(la::BaseLaplace)
         params = [θ for θ in params]                                         # get all parameters and constants in logitbinarycrossentropy
     elseif la.subset_of_weights == :last_layer
         params = [params[n_elements - 1], params[n_elements]]                # only get last layer parameters and constants
+    elseif la.subset_of_weights == :sub_network
+        indeces = la.sub_network_indices
+        params = [params[i] for i in indeces]                                # only get parameters and constants of the sub-network
     end                                                                      # params[n_elements-1] is the weight matrix of the last layer
```

Comment:
Thank you, this is very helpful insight! In this case, what do you think subnetwork_indices would look like in Julia?
And in regards to the performance advantage of this, according to the Laplace Redux paper, it should help with storing the covariance matrix for very large neural networks. However, I believe they also recognize this technique to not be the most helpful, because finding an optimal subnetwork involves combinatorial computations. So my short answer to this would be - I honestly have no clue.:)

Reply:
> In this case, what do you think subnetwork_indices would look like in Julia?
So at the internal level, we will represent these as an array (vector / AbstractArray) of *flattened* indices, id est integers corresponding to the flattened representation of theta (all model parameters in one vector, just like `torch.nn.utils.parameters_to_vector` does it).
Yes, PA said having a smaller J would help for faster multiplication.
> finding an optimal subnetwork involves combinatorial computations.
For now, this is something we leave to the user. There is a technique referenced in Daxberger et al. of selecting these parameters to optimize some metric, like marginal variance